### PR TITLE
added meta info for `block-indentation` rule

### PR DESCRIPTION
**Pull request by MelSumner**
If this is how we want to add the meta info for each rule, I'll do this for each rule. 

The idea is that adding meta info to each rule will make it easier to break out the documentation in a more automated fashion.

### CODE REVIEW

**Review by rwjblue — APPROVED** [review]
Seems good to me, @bmish what do you think?

File: lib/rules/block-indentation.js [hunk 1]
```diff
@@ -82,6 +82,16 @@ function isControlChar(char) {
...
 
 module.exports = class BlockIndentation extends Rule {
+  static get meta() {
+    return {
+      description: 'requires blocks to be indented and aligned',
+      category: 'Stylistic Issues', // 'Deprecated Rules', 'Possible Error', 'Best Practices',
+      presets: { recommended: true },
+      url:
+        'https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rules/block-indentation.md',
+      fixable: false,
```
Comment (bmish) [inline lib/rules/block-indentation.js:92]: I would like to exactly match the [eslint rule meta object](https://eslint.org/docs/developer-guide/working-with-rules#rule-basics-1):

```suggestion
      type: 'layout',
      docs: {
        description: 'requires blocks to be indented and aligned',
        category: 'Stylistic Issues', // 'Deprecated Rules', 'Possible Error', 'Best Practices',
        recommended: false,
        url: 'https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rules/block-indentation.md',
      },
      fixable: false,
      schema: [],
```

File: lib/rules/block-indentation.js [hunk 2]
```diff
@@ -82,6 +82,16 @@ function isControlChar(char) {
 }
 
 module.exports = class BlockIndentation extends Rule {
+  static get meta() {
```
Comment (Turbo87) [inline lib/rules/block-indentation.js:85]: any reason for this being a property method instead of something like `static meta = { ... }`?

Comment (rwjblue) [reply lib/rules/block-indentation.js:85]: I don't think static class fields are supported on our supported Node versions. Could be wrong though...

**Pull request closed without merging**